### PR TITLE
feat: Fixes issue #385: Add `showWeekends` flag in month view

### PR DESCRIPTION
**Pull request by shubham-jitiya-simform**
# Description

- Fixed if `daysNotInMonth` true hide events on hidden date.
- Added `showWeekends` flag to hide weekends. Default value is true.
- Removed empty rows if entire week dates doesn't fall in current month.
- Updated readme & change log
- Attached GIF which shows example.

ShowWeekends flag

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [x] Yes, this PR is a breaking change.
- [ ] No, this PR is not a breaking change.

## Related Issues

Closses #385 

[Contributor Guide]: https://github.com/SimformSolutionsPvtLtd/flutter_calendar_view/blob/master/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# [1.4.0](https://github.com/SimformSolutionsPvtLtd/flutter_calendar_view/tree/1.4.0)
+- Adds `showWeekends` flag in month view to hide & show weekends view. 
+  Default is `showWeekends = true` shows all weekdays. [#385](https://github.com/SimformSolutionsPvtLtd/flutter_calendar_view/issues/385)
+- Fixes issue: Event is shown when days not in current month is hidden `hideDaysNotInMonth = true`.
```
Comment (ParthBaraiya) [inline CHANGELOG.md:4]: can we improve this statement.

File: lib/src/extensions.dart [hunk 2]
```diff
@@ -87,11 +87,37 @@ extension DateTimeExtensions on DateTime {
...
   /// which will contain 6 weeks that is the maximum number of weeks a month
   /// can have.
-  List<DateTime> datesOfMonths({WeekDays startDay = WeekDays.monday}) {
+  ///
+  /// It excludes week if `hideDaysNotInMonth` is set true and
+  /// if all dates in week comes in next month then it will excludes that week.
+  List<DateTime> datesOfMonths({
+    WeekDays startDay = WeekDays.monday,
+    bool hideDaysNotInMonth = false,
+    bool showWeekends = false,
```
Comment (ParthBaraiya) [inline lib/src/extensions.dart:96]: default value of showWeekends should be true.

File: lib/src/month_view/month_view.dart [hunk 3]
```diff
@@ -64,6 +56,14 @@ class MonthView<T extends Object?> extends StatefulWidget {
   /// This method will be called when user double taps on event tile.
   final TileTapCallback<T>? onEventDoubleTap;
 
+  /// Show weekends or not.
+  /// Default value is true.
+  ///
+  /// Similar to [WeekView],
+  /// If it is false week view will remove weekends from week days
```
Comment (ParthBaraiya) [inline lib/src/month_view/month_view.dart:63]: remove the mention of week view.

File: lib/src/month_view/month_view.dart [hunk 4]
```diff
@@ -391,6 +398,7 @@ class MonthViewState<T extends Object?> extends State<MonthView<T>> {
                               startDay: widget.startDay,
                               physics: widget.pagePhysics,
                               hideDaysNotInMonth: widget.hideDaysNotInMonth,
+                              weekDays: widget.showWeekends ? 7 : 5,
                             ),
                           );
                         }),
```
Comment (ParthBaraiya) [inline lib/src/month_view/month_view.dart:404]: add , and reformat file.

File: lib/src/month_view/month_view.dart [hunk 5]
```diff
@@ -724,6 +744,32 @@ class _MonthPageBuilder<T> extends StatelessWidget {
...
+  /// if the `weekDays` parameter is set to 5, ensuring only
+  /// weekdays are included in the returned list.
+  List<DateTime> get _filteredDays {
+    final monthDays = date.datesOfMonths(
+      startDay: startDay,
+      hideDaysNotInMonth: hideDaysNotInMonth,
+      showWeekends: weekDays == 7,
+    );
+
+    return monthDays.where((day) {
```
Comment (ParthBaraiya) [inline lib/src/month_view/month_view.dart:764]: can we move this logic in datesOfMonths as we are already passing the flag in the method.

**Review by ParthBaraiya — CHANGES_REQUESTED** [review]

File: lib/src/month_view/month_view.dart (hunk 5 above)
Comment (shubham-jitiya-simform) [reply lib/src/month_view/month_view.dart]: Removed this and used `take()` to take only the required dates from the list of week. 7 for Week ends and 5 with hidden weekends.

Comment (ParthBaraiya) [reply lib/src/month_view/month_view.dart]: take will cause issue if startDay is otherthan Monday.
Screenshot 2024-12-19 004237

**Review by ParthBaraiya — CHANGES_REQUESTED** [review]
Use where instead of take. take will cause issue if startDay if other than monday.

**Review by PRBaraiya — APPROVED** [review]
LGTM!

**Review by ParthBaraiya — APPROVED** [review]

**Comment by ParthBaraiya** [comment]
RIght now, the github action is giving error which should be fixed in dark theme support.

**Pull request merged**